src/android/FileTransferBackground.java

```diff
@@ -126,23 +140,52 @@ public boolean execute(String action, JSONArray args, final CallbackContext call
         return true;
     }
 
+    private void createNotificationChannel() {
+        if (Build.VERSION.SDK_INT >= 26) {
+            NotificationChannel channel = new NotificationChannel(
+                    "com.spoon.backgroundfileupload.channel",
+                    "upload channel",
+                    NotificationManager.IMPORTANCE_LOW
+            );
+            NotificationManager manager = (NotificationManager) cordova.getActivity()
+                    .getApplication()
+                    .getApplicationContext()
+                    .getSystemService(Context.NOTIFICATION_SERVICE);
+            manager.createNotificationChannel(channel);
+        }
+    }
 
     private void initManager(String options) throws IllegalStateException {
         if (this.ready) {
             throw new IllegalStateException("initManager was called twice");
         }
         this.ready = true;
+        String notificationChannelID = "com.spoon.backgroundfileupload.channel";
+        UploadServiceConfig.initialize(
+                this.cordova.getActivity().getApplication(),
+                notificationChannelID,
+                false
+        );
+        this.globalObserver = new RequestObserver(this.cordova.getActivity().getApplicationContext(), broadcastReceiver);
+        this.globalObserver.register();
         int parallelUploadsLimit = 1;
         try {
             JSONObject settings = new JSONObject(options);
             parallelUploadsLimit = settings.getInt("parallelUploadsLimit");
         } catch (JSONException error) {
             logMessage("eventLabel='could not read parallelUploadsLimit from config' error='" + error.getMessage() + "'");
         }
-        UploadService.HTTP_STACK = new OkHttpStack();
-        UploadService.UPLOAD_POOL_SIZE = parallelUploadsLimit;
-        UploadService.NAMESPACE = cordova.getContext().getPackageName();
-        cordova.getActivity().getApplicationContext().registerReceiver(broadcastReceiver, new IntentFilter(UploadService.NAMESPACE + ".uploadservice.broadcast.status"));
+        UploadServiceConfig.setHttpStack(new OkHttpStack());
+        ExecutorService threadPoolExecutor =
+                new ThreadPoolExecutor(
+                        parallelUploadsLimit,
+                        parallelUploadsLimit,
```

Comment:
Est-ce que ça (maximumPoolSize) doit être parallelUploadsLimit ? Une meilleur limit me parait mieux... 50 ? 100 ?

> If a request cannot be queued, a new thread is created unless this would exceed maximumPoolSize, in which case, the task will be rejected.
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html

Reply:
Vu avec @azharbeebeejaun c'est ok